analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -49,27 +49,17 @@ public abstract class MethodsShouldNotHaveIdenticalImplementationsBase<TSyntaxKi
                         return;
                     }
 
-                    var methods = GetMethodDeclarations(c.Node).ToList();
+                    var methodsToHandle = new LinkedList<TMethodDeclarationSyntax>(GetMethodDeclarations(c.Node));
 
-                    var alreadyHandledMethods = new HashSet<TMethodDeclarationSyntax>();
-
-                    foreach (var method in methods)
+                    while (methodsToHandle.Count > 0)
```

Comment:
Convention in the codebase
```suggestion
                    while (methodsToHandle.Any())
```